mne/stats/regression.py

```diff
@@ -392,3 +392,254 @@ def _make_evokeds(coefs, conds, cond_length, tmin_s, tmax_s, info):
+            The estimator to use for fitting. This is any object that contains
+            a `fit` and `predict` method, which takes inputs of the form
+            (X, y), and which creates a `.coef_` attribute upon fitting.
+            This may be a pipeline, in which case the final estimator must
+            follow the same conventions described above.
+
+        References
+        ----------
+        .. [1] Smith, N. J., & Kutas, M. (2015). Regression-based estimation of
+               ERP waveforms: II. Non-linear effects, overlap correction, and
+               practical considerations. Psychophysiology, 52(2), 169-189.
+        .. [2] Theunissen, F. E. et al. Estimating spatio-temporal receptive
+               fields of auditory and visual neurons from their responses to
+               natural stimuli. Network 12, 289-316 (2001).
+        .. [3] Willmore, B. & Smyth, D. Methods for first-order kernel
+               estimation: simple-cell receptive fields from responses to
+               natural scenes. Network 14, 553-77 (2003).
+        """
+        self.delays = np.array([0]) if delays is None else delays
+        self.n_delays = len(self.delays)
+        self.est = _check_estimator(est)
+        self.preproc_y = preproc_y
+
+    def fit(self, raw, continuous=None, events=None, event_id=None,
+            picks=None, fit_ixs=None, continuous_names=None, verbose=False):
+        """Fit the model.
+
+        Fits a receptive field model. Model results are stored as attributes.
+
+        Parameters
+        ----------
+        raw : instance of MNE Raw
+            The data on which we want to fit a regression model.
+        events : array, shape (n_events, 3)
+            An MNE events array specifying indices for event onsets
+        event_id : dictionary | None
+            A dictionary of (event_name: event_int) pairs, corresponding to the
+            mapping from event name strings to the 3rd column of events.
+        continuous : array, shape (n_feats, n_times)
+            Continuous input features in the regression.
+        continuous_names : array, shape (n_feats,) | None
+            Names for the input continuous variables.
+        picks : array, shape (n_picks,) | None
+            Indices for channels to use in model fitting. If None, all channels
+            will be fit.
+        fit_ixs : array, dtype int | None
+            An array specifying indices of timepoints to use in the regression.
+            Useful if a subset of datapoints is needed but lags must be
+            calculated on the whole input.
+        verbose : bool
+            If True, will display a progress bar during fits for CVs remaining.
+
+        Attributes
+        ----------
+        coefs_ : array, shape (n_features, n_lags, len(picks))
+            The average coefficients across CV splits
+        coefs_all_ : array, shape(n_cv, n_features * n_lags)
+            The raw coefficients for each iteration of cross-validation.
```

Comment:
If I understand correctly, the CV in these kind of scenario is intrinsically invalid because of the time dependence across samples (e.g. a channel may be bad during the first half of the experiment).

If this is correct, then I would leave the CV issue outside of this object for now, and make an explicit 2 fold division validation (first vs second half of recordings) in the example.